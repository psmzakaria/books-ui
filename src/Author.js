import React, { Component } from 'react';
const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";
class Author extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }
  componentDidMount() {
    this.getAuthors("");
  }
  getAuthors = async search => {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    const authorsData = await response.json();
    this.setState({ authors: authorsData });
  };

  render() {
    return (
      <div>
        {this.state.authors.map((author, i) => {
          return <li> {author.name}</li>;
        })}
      </div>
    );
  }
}

export default Author;
