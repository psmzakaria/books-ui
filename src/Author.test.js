import React from "react";
import ReactDOM from "react-dom";
import Author from "./Author";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders title and authorlist after the component mount", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        name: "roald Dhal",
        age: 29,
        _id:"47389577"
      },
      {
        name: "Dolly Lama",
        age: 99,
        _id:"473895789547"
      }
    ])
  );

  const renderer = new ShallowRenderer();
  renderer.render(<Author />);
  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
