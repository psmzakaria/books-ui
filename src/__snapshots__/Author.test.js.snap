// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title and authorlist after the component mount 1`] = `
<div>
  
</div>
`;
